### PR TITLE
Fix for histogram merging

### PR DESCRIPTION
**Pull request by ksneab7**

### CODE REVIEW

File: dataprofiler/profilers/histogram_utils.py [hunk 1]
```diff
@@ -14,6 +14,218 @@
 )
 
 
+def rework_ptp(maximum, minimum):
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:17]: once we fix naming, we should make these private as well.

File: dataprofiler/profilers/histogram_utils.py [hunk 2]
```diff
@@ -14,6 +14,218 @@
...
+    ----------
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:45]: We can instead use the `profile.sampling_size`.

File: dataprofiler/profilers/histogram_utils.py [hunk 3]
```diff
@@ -14,6 +14,218 @@
...
+    minimum = profile._stored_histogram["histogram"]["bin_edges"][0]
+    maximum = profile._stored_histogram["histogram"]["bin_edges"][-1]
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:47]: Should be able to use `profile.min` etc.

File: dataprofiler/profilers/histogram_utils.py [hunk 4]
```diff
@@ -14,6 +14,218 @@
...
+    return (rework_ptp(maximum, minimum) /
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:81]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 5]
```diff
@@ -14,6 +14,218 @@
...
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
+    minimum = profile._stored_histogram["histogram"]["bin_edges"][0]
+    maximum = profile._stored_histogram["histogram"]["bin_edges"][-1]
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:106]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 6]
```diff
@@ -14,6 +14,218 @@
...
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:129]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 7]
```diff
@@ -14,6 +14,218 @@
...
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    iqr = np.subtract(profile._get_percentile([75]), profile._get_percentile([25]))
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:159]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 8]
```diff
@@ -14,6 +14,218 @@
...
+    ----------
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:224]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 9]
```diff
@@ -90,3 +302,42 @@ def _get_bin_edges(
...
+    if isinstance(bin_method, str):
+        bin_name = bin_method
+    # if `bins` is a string for an automatic method,
+    # this will replace it with the number of bins calculated
+    if bin_name not in _hist_bin_selectors:
+        raise ValueError(f"{bin_name!r} is not a valid estimator for `bins`")
+
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
+    minimum = profile._stored_histogram["histogram"]["bin_edges"][0]
+    maximum = profile._stored_histogram["histogram"]["bin_edges"][-1]
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:328]: as above

File: dataprofiler/profilers/histogram_utils.py [hunk 10]
```diff
@@ -90,3 +302,42 @@ def _get_bin_edges(
...
+
+    if dataset_size == 0:
+        n_equal_bins = 1
+    else:
+        # Do not call selectors on empty arrays
+        width = rework_hist_bin_selectors[bin_name](profile)
```
[the remark this answers is not in the document]
Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:334]: I'm curious if there are ways to reduce overall calcs by generating props before this loop and passing that in instead of hte profile.

File: dataprofiler/profilers/numerical_column_stats.py [hunk 11]
```diff
@@ -76,7 +76,6 @@ def __init__(self, options: NumericalOptions = None) -> None:
         self.histogram_selection: str | None = None
         self.user_set_histogram_bin: int | None = None
         self.bias_correction: bool = True  # By default, we correct for bias
-        self._mode_is_enabled: bool = True
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:79]: Why is this removed?

File: dataprofiler/profilers/histogram_utils.py (hunk 2 above)
Comment (ksneab7) [reply dataprofiler/profilers/histogram_utils.py:45]: AttributeError: 'IntColumn' object has no attribute 'sampling_size'

File: dataprofiler/profilers/numerical_column_stats.py (hunk 11 above)
Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py:79]: Weird I think this may solve some of my issue above. This should not be deleted

File: dataprofiler/profilers/histogram_utils.py [hunk 12]
```diff
@@ -90,3 +366,61 @@ def _get_bin_edges(
...
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+    :param bin_method: the method used to calculate bins based on the profile given
+    :type bin_method: string
+
+    :return: ideal number of bins for a particular histograom calulcation method
+    """
+    # parse the overloaded bins argument
+
+    rework_hist_bin_selectors = {
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:384]: instead of re init internal every time, maybe global in this file. Probably should privatize the variable too. Update name to suggest from profile in the name as oppose to rework.

File: dataprofiler/profilers/histogram_utils.py [hunk 13]
```diff
@@ -14,6 +14,293 @@
 )
 
 
+def _ptp(maximum, minimum):
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:17]: do we need all of the imports above? also, do we know why we `ignore`?

File: dataprofiler/profilers/histogram_utils.py [hunk 14]
```diff
@@ -14,6 +14,293 @@
...
+    """
+    Doane's histogram bin estimator reworked to use profiles.
+
+    Improved version of Sturges' formula which works better for
+    non-normal data. See
+    stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    Parameters
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:39]: I'm thinking we should reformat to use our format for docstrings until we want to convert to a different technique.

File: dataprofiler/profilers/histogram_utils.py [hunk 15]
```diff
@@ -14,6 +14,293 @@
...
+        Input data's data profile that is to be histogrammed
+
+    Returns
+    -------
+    An estimate of the optimal bin width for the given data.
+    """
+    try:
+        dataset_size = profile.match_count
+    except AttributeError:
+        dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:51]: We do this a lot. Maybe abstract to a function we use at the top instead?

File: dataprofiler/profilers/histogram_utils.py [hunk 16]
```diff
@@ -14,6 +14,293 @@
...
+
+    minimum = (
+        profile.min
+        if profile.min is not None
+        else profile._stored_histogram["histogram"]["bin_edges"][0]
+    )
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:107]: similar above for re-use. If we do a func, privatize.

File: dataprofiler/profilers/histogram_utils.py [hunk 17]
```diff
@@ -14,6 +14,293 @@
...
+    "auto": _calc_auto_bin_width_from_profile,
+    "doane": _calc_doane_bin_width_from_profile,
+    "fd": _calc_fd_bin_width_from_profile,
+    "rice": _calc_rice_bin_width_from_profile,
+    "scott": _calc_scott_bin_width_from_profile,
+    "sqrt": _calc_sqrt_bin_width_from_profile,
+    "sturges": _calc_sturges_bin_width_from_profile,
+}
+
+
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:303]: at this point in the code, have already calc'ed the values necessary to generate this? as if we did it from the profile? This would be __calculations order dependent.

Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:303]: could be a refactor to remove.

File: dataprofiler/profilers/histogram_utils.py (hunk 13 above)
Comment (ksneab7) [reply dataprofiler/profilers/histogram_utils.py:17]: We do use all of the imports above in the `get_bin_edges` function and no not sure why there is an ignore, probably because of reference to protected methods?

File: dataprofiler/profilers/histogram_utils.py (hunk 17 above)
Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:303]: nvm, maybe we cannot bc initial of hist not yet created for quantiles?

File: dataprofiler/profilers/histogram_utils.py (hunk 14 above)
Comment (ksneab7) [reply dataprofiler/profilers/histogram_utils.py:39]: I am not sure. this may depend on Justin's feedback so Ill wait it out for now

File: dataprofiler/profilers/histogram_utils.py [hunk 18]
```diff
@@ -90,3 +377,49 @@ def _get_bin_edges(
...
+        else profile._stored_histogram["histogram"]["bin_edges"][-1]
+    )
+
+    if dataset_size == 0:
+        n_equal_bins = 1
+    else:
+        # Do not call selectors on empty arrays
+        width = _hist_bin_width_selectors_for_profile[bin_method](profile)
+        if width and not np.isnan(width):
+            n_equal_bins = int(np.ceil(_unsigned_subtract(maximum, minimum) / width))
```
Comment (JGSweets) [inline dataprofiler/profilers/histogram_utils.py:420]: how is this `_unsigned...` diff from the `_ptp` above? Case for re-use?

File: dataprofiler/profilers/numerical_column_stats.py [hunk 19]
```diff
@@ -179,23 +179,87 @@ def _add_helper_merge_profile_histograms(
...
                 "current_loss": 0,
                 "histogram": {"bin_counts": None, "bin_edges": None},
             }
 
-        combined_values = np.concatenate(
-            [other1._histogram_to_array(), other2._histogram_to_array()]
+        # calculate the min of the first edge and the max of the last edge
+        # between two arrays
+        if not (other1.min is None or other2.min is None):
+            global_min_of_histogram_edges = min(other1.min, other2.min)
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:198]: bc this is an or, what happens if one of them is `None`. does this error?/

File: dataprofiler/profilers/histogram_utils.py (hunk 15 above)
Comment (ksneab7) [reply dataprofiler/profilers/histogram_utils.py:51]: I think the correct way to do it would be to calculate it one time and use it like how I had it in the temporary rework I indicated previously. We can turn it into a function temporarily but I think the actual fix will come later when we make this code work together with the pre-existing code such that this number will only need to be calulated one time.

File: dataprofiler/profilers/numerical_column_stats.py (hunk 19 above)
Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py:198]: if one them is None the whole expression is false so no it does not. If the expression was 
`not (other1.min is None and other2.min is None)` we may have an issue

File: dataprofiler/profilers/numerical_column_stats.py [hunk 20]
```diff
@@ -179,23 +179,87 @@ def _add_helper_merge_profile_histograms(
...
+        else:
+            global_min_of_histogram_edges = min(
+                other1._stored_histogram["histogram"]["bin_edges"][0],
+                other2._stored_histogram["histogram"]["bin_edges"][0],
+            )
+
+        if not (other1.max is None or other2.max is None):
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:205]: as above

File: dataprofiler/profilers/numerical_column_stats.py (hunk 19 above)
Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:198]: at this point we have the `self.min` I think since we moved the calc below.

We could do:
```python
global_min_of_histogram_edges = self.min
if self.min is None:
    global_min_of_histogram_edges = min(
        other1._stored_histogram["histogram"]["bin_edges"][0],
        other2._stored_histogram["histogram"]["bin_edges"][0],
    )
```
it removes the `else` and utilizes existing functionality.

This is conditional on `self.min` being updated prior

File: dataprofiler/profilers/numerical_column_stats.py [hunk 21]
```diff
@@ -1240,11 +1305,43 @@ def _regenerate_histogram(
...
             )
+        return self._assimilate_histogram(
+            entity_count_by_bin,
+            bin_edges,
+            new_bin_counts,
+            new_bin_edges,
+            suggested_bin_count,
+        )
 
+    def _assimilate_histogram(
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:1316]: can this be a class method? does it need to use `self`?

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:1316]: We would have to get around this:
```python
is_float_profile = self.__class__.__name__ == "FloatColumn"
```

File: dataprofiler/profilers/numerical_column_stats.py [hunk 22]
```diff
@@ -1240,11 +1305,43 @@ def _regenerate_histogram(
...
+        self,
+        entity_count_by_bin: np.ndarray,
+        bin_edges: np.ndarray,
+        new_entity_count_by_bin: np.ndarray,
+        new_bin_edges: np.ndarray,
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:1321]: Trying to think how we can bring clarity to the word assimilate. instead of `new`, maybe `destination`

File: dataprofiler/profilers/text_column_profile.py [hunk 23]
```diff
@@ -56,8 +56,8 @@ def __add__(self, other: TextColumn) -> TextColumn:
                 "'TextColumn' and '{}'".format(other.__class__.__name__)
             )
         merged_profile = TextColumn(None)
-        NumericStatsMixin._add_helper(merged_profile, self, other)
         BaseColumnPrimitiveTypeProfiler._add_helper(merged_profile, self, other)
+        NumericStatsMixin._add_helper(merged_profile, self, other)
```
Comment (JGSweets) [inline dataprofiler/profilers/text_column_profile.py:60]: I need to think about how this order could have affected other things, but tests do pass.

File: dataprofiler/tests/profilers/test_histogram_utils.py [hunk 24]
```diff
@@ -0,0 +1,348 @@
...
+            "dataprofiler.profilers.NumericStatsMixin.stddev", new_callable=mock_stddev
+        ):
+            # Case 1: min, max, match_count, biased_skewness, and stddev are set
+            sg1 = np.sqrt(6.0 * (5 - 2) / ((5 + 1.0) * (5 + 3)))
+            expected = histogram_utils._ptp(5, 1) / (
+                1.0 + np.log2(5) + np.log2(1.0 + np.absolute(1.0) / sg1)
+            )
+
+            actual = histogram_utils._calc_doane_bin_width_from_profile(profile)
+            self.assertAlmostEqual(expected, actual, places=9)
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_histogram_utils.py:91]: do we need to almost equals after the update to the calc?

File: dataprofiler/tests/profilers/test_histogram_utils.py [hunk 25]
```diff
@@ -0,0 +1,348 @@
...
+
+    def test_calc_doane_bin_width_from_profile(self):
+        # Initial setup of profile
+        profile = TestColumn()
+
+        with mock.patch(
+            "dataprofiler.profilers.NumericStatsMixin.stddev", new_callable=mock_stddev
+        ):
+            # Case 1: min, max, match_count, biased_skewness, and stddev are set
+            sg1 = np.sqrt(6.0 * (5 - 2) / ((5 + 1.0) * (5 + 3)))
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_histogram_utils.py:85]: Should we replace the values themselves with the settings above?

e.g. `1-> self.min`

File: dataprofiler/tests/profilers/test_histogram_utils.py [hunk 26]
```diff
@@ -0,0 +1,348 @@
...
+        # Initial setup of profile
+        profile = TestColumn()
+
+        # Case 1: bin method not in set of valid bin methods
+        with self.assertRaises(ValueError):
+            histogram_utils._calculate_bins_from_profile(profile, "test_not_in_method")
+
+        # Case 2: min, max, and match_count are set and bin_method is valid
+        expected_sqrt_answer = 4 / 2.23606797749979
+        expected_buckets = int(np.ceil(4 / expected_sqrt_answer))
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_histogram_utils.py:302]: for this, could probably just set the int value as opposed to the entire calc.

File: dataprofiler/profilers/histogram_utils.py (hunk 13 above)
Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:17]: we could update the ignore to specifically capture that condition as oppose to a totality

File: dataprofiler/profilers/histogram_utils.py (hunk 15 above)
Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:51]: Agreed, a better system. But, until then probably still ideal to nav towards dry. no extra tests needed.

File: dataprofiler/profilers/numerical_column_stats.py (hunk 19 above)
Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py]: else is needed. mypy fails because of  requirements from lispace parameter types

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py]: We shouldn't be changing code to meet mypy expectations. I'd defer to cleanliness and readability over mypy, but it typically is a good guide towards that end.

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py]: It failing from that change also would indicate an issue in our description / typing else where.

File: dataprofiler/profilers/numerical_column_stats.py [hunk 27]
```diff
@@ -179,23 +179,89 @@ def _add_helper_merge_profile_histograms(
...
                 "current_loss": 0,
                 "histogram": {"bin_counts": None, "bin_edges": None},
             }
 
-        combined_values = np.concatenate(
-            [other1._histogram_to_array(), other2._histogram_to_array()]
+        # calculate the min of the first edge and the max of the last edge
+        # between two arrays
+        global_min_of_histogram_edges = (
+            float(self.min)
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:198]: if self.min is None, this will error

File: dataprofiler/profilers/numerical_column_stats.py (hunk 19 above)
Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py]: I figured it out, it was bc i wrote `if self.min is None` needs to be `if global_min_of_histogram_edges is None`

```python
global_min_of_histogram_edges = self.min
        if global_min_of_histogram_edges is None:
            global_min_of_histogram_edges =  min(
                other1._stored_histogram["histogram"]["bin_edges"][0],
                other2._stored_histogram["histogram"]["bin_edges"][0],
            )
```

File: dataprofiler/profilers/text_column_profile.py (hunk 23 above)
Comment (ksneab7) [reply dataprofiler/profilers/text_column_profile.py:60]: So, im not sure if there was a deliberate reason to have this order not match the other profiles that use these two functions, but  the other two (float and int) do it in the updated order you see in the change. Previously order didn't matter so that might have been why it differed and no one said anything in the PR.

Comment (ksneab7) [reply dataprofiler/profilers/text_column_profile.py:60]: Looking farther into this, this specific line was implemented on release 3 years ago, so I wouldn't really know the reasoning behind the order differing decision as thats before my time...

File: dataprofiler/profilers/numerical_column_stats.py (hunk 21 above)
Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:1318]: probably unnecessary right now.

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/histogram_utils.py [hunk 28]
```diff
@@ -90,3 +311,37 @@ def _get_bin_edges(
         raise ValueError("`bins` must be 1d, when an array")
 
     return bin_edges, n_equal_bins
+
+
+def _calculate_bins_from_profile(profile, bin_method):
+    """
+    Compute the bins used internally by `histogram`.
```
[the remark this answers is not in the document]
Comment (ksneab7) [reply dataprofiler/profilers/histogram_utils.py:318]: This doesnt have a numpy equivalent. So no I dont think so

**Review by taylorfturner — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/numerical_column_stats.py [hunk 29]
```diff
@@ -179,23 +179,93 @@ def _add_helper_merge_profile_histograms(
...
+            calculated_loss = (
+                (histogram_loss_1 * other1.sample_size)
+                + (histogram_loss_2 * other2.sample_size)
+            ) / (other1.sample_size + other2.sample_size)
+        except AttributeError:
+            sample_size_1 = sum(other1._stored_histogram["histogram"]["bin_counts"])
+            sample_size_2 = sum(other2._stored_histogram["histogram"]["bin_counts"])
+            calculated_loss = (
+                (histogram_loss_1 * sample_size_1) + (histogram_loss_2 * sample_size_2)
+            ) / (sample_size_1 + sample_size_2)
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:256]: Is `sample_size` the correct value or should it be `match_count`? Do we have tests evaluating this?

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:256]: Also, I think loss should be the sum, right? It's the aggregate loss of both being inserted into the `_stored_histogrtam`.

File: dataprofiler/profilers/numerical_column_stats.py [hunk 30]
```diff
@@ -179,23 +179,93 @@ def _add_helper_merge_profile_histograms(
...
+        _, histogram_loss_2 = self._assimilate_histogram(
+            from_hist_entity_count_per_bin=other2._stored_histogram["histogram"][
+                "bin_counts"
+            ],
+            from_hist_bin_edges=other2._stored_histogram["histogram"]["bin_edges"],
+            dest_hist_entity_count_per_bin=new_entity_count_by_bin,
+            dest_hist_bin_edges=ideal_bin_edges,
+            dest_hist_num_bin=ideal_count_of_bins,
+        )
+        try:
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:246]: we could separate out getting sizes from the actual calc so the calc is not within the try itself. that is if sizes are needed.

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:246]: see below

File: dataprofiler/profilers/numerical_column_stats.py (hunk 29 above)
Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py:256]: sure match_count is probably right.
so in this calculation we are taking the the loss of each histogram and using the size of the dataset associated with that loss as a weight factor. So yes it is a sum, in this case I am doing a weighted sum.

File: dataprofiler/profilers/numerical_column_stats.py [hunk 31]
```diff
@@ -1280,23 +1377,30 @@ def _regenerate_histogram(
...
                 ) ** 2 * count_in_left_bin
 
                 # allocate leftovers to the right bin
-                new_bin_counts[new_bin_id + 1] += bin_count - count_in_left_bin
+                dest_hist_entity_count_per_bin[new_bin_id + 1] += (
+                    bin_count - count_in_left_bin
+                )
                 hist_loss += (
-                    ((new_bin_edge[2] - new_bin_edge[1]) - (bin_edge[1] - bin_edge[0]))
+                    ((new_bin_edge[2] + new_bin_edge[1]) - (bin_edge[1] + bin_edge[0]))
```
Comment (ksneab7) [inline dataprofiler/profilers/numerical_column_stats.py:1391]: This was the last bug to be fixed

**Review by taylorfturner — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

File: dataprofiler/profilers/histogram_utils.py [hunk 32]
```diff
@@ -1,19 +1,241 @@
...
+    "fd": _calc_fd_bin_width_from_profile,
+    "rice": _calc_rice_bin_width_from_profile,
+    "scott": _calc_scott_bin_width_from_profile,
+    "sqrt": _calc_sqrt_bin_width_from_profile,
+    "sturges": _calc_sturges_bin_width_from_profile,
+}
+
+
 def _get_bin_edges(
     a: np.ndarray,
```
Comment (tyfarnan) [inline dataprofiler/profilers/histogram_utils.py:240]: think we might like a more descriptive variable name here?

Comment (JGSweets) [reply dataprofiler/profilers/histogram_utils.py:240]: A future fix, not in code we changed and is actually how `numpy` also implements.

Comment (taylorfturner) [reply dataprofiler/profilers/histogram_utils.py:240]: good call

**Review by tyfarnan — APPROVED** [review]
